aiohttp/web_urldispatcher.py

```diff
@@ -93,7 +93,13 @@ def __init__(self, method, handler, *,
               issubclass(handler, AbstractView)):
             pass
         else:
-            handler = asyncio.coroutine(handler)
+            @asyncio.coroutine
+            def handler_wrapper(*args, **kwargs):
+                result = handler(*args, **kwargs)
+                if asyncio.iscoroutine(result):
```

Comment:
according to current documentation (for UrlDispatcher.add_route and Resource.add_route) handler should be callable which excludes asyncio.Future